workspace/update-injected-packages/test/DirPatcher.test.ts

```diff
@@ -181,10 +181,11 @@ test('optimally synchronizes source and target', async () => {
   expect(fsMethods.mkdir).toHaveBeenCalledWith(path.resolve(targetDir, 'files-to-add'), expect.anything())
   expect(fsMethods.mkdir).toHaveBeenCalledWith(path.resolve(targetDir, 'files-to-add/a'), expect.anything())
 
-  // Windows CI fails this step because of exactly one call to `target`.
-  // Not sure why it was called.
-  if (process.platform !== 'win32') {
-    // reuses `stat` results from @pnpm/directory-fetcher
+  // reuses `stat` results from @pnpm/directory-fetcher
+  if (process.platform === 'win32') {
+    // Some code somewhere may have called `stat` on `target` exactly once.
```

Comment:
Could it be this line?

https://github.com/pnpm/pnpm/blob/428915cd9dcfe8c9a3c10c0bc2651dc488d0e828/pkg-manifest/read-project-manifest/src/index.ts#L101

Reply:
I don't know. You don't have windows? I can check later?

Reply:
That line of code is getting called.

Reply:
Well, it turns out I was wrong to test `stat` reuses in `Patcher.apply()` because the reuse happens in `Patcher.fromMultipleTargets` which calls both `fetchFromDir` (which provides the stats) and `extendFilesMap` (which would reuses the stats).

https://github.com/pnpm/pnpm/pull/9100/commits/9bed5d8cb698bfdc72750f387225f23e61679505